### DISCUSSION TITLE
Midi devices

### DISCUSSION THREAD

**Discussion by spamatica** [muse-sequencer/muse#1214] [Team Posts]
Midi configuration in MusE has a lot of cool features but sometimes I end up in situations that are curious.
There was a question on the forum yesterday about too many jack-devices, not entirely clear what the problem was, the guy is not always so easy to understand...
Today, trying some stuff while doing documentation I ended up in the situation below (which may be what they meant).
Now, I'm not 100% sure I should blame MusE here but I cannot remember any other applications I have run.
As can be seen below, I have, for some reason 16 ALSA midi through devices. Does MusE at any point modify such devices or is it something else?

![image](https://user-images.githubusercontent.com/4476983/115764510-ef9e6a80-a3a5-11eb-9c13-6c8ac94d5b1d.png)

**Comment by terminator356** [comment]
Phht. What the...?
OK you're going to have to give me more to go on here.
If you can recreate it, please open something like QJackCtl and look in the ALSA midi tab
 and tell me if they are listed there as well.
I mean... we just follow what ALSA tells us is there.
Especially since the names you see on the left are **not** ones that we give the ports,
 they are the actual ALSA names on each port. We don't make up those names.
So that's a **big clue** that your system really does, or did, have 16 ALSA through ports.

**Comment by spamatica** [comment]
Maybe I was connecting dots that didn't belong together.

I have no idea where the devices come from but they exist already when the machine is cold booted (if I check with for instance qjackctl) so it doesn't look like MusE was involved.
